printer/scraperPrinter/influxHandler.js

```diff
@@ -0,0 +1,82 @@
+const fetch = require('node-fetch');
+const {HpLaserJetP2015} = require('../snmp.js');
+const logger = require('../logger.js');
+class influxHandler{
+
+  constructor(printerIP, influxBaseUrl, printerName) {
+    this.snmpHandler = new HpLaserJetP2015(printerIP);
+    this.influxBaseUrl = influxBaseUrl;
+    this.printerName = printerName;
+  }
+
+  /**
+   * Creates DataBase by the name of quasar_data if not made already
+   */
+  async initializeInfluxDb() {
+    logger.warn('Creating influx Database called quasar_data if not created already');
+    try{
+      await fetch(`${this.influxBaseUrl}/query`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        body: 'q=CREATE DATABASE "quasar_data"'
+      });
+    }catch(error){
+      logger.error('Error Creating Database!');
+    }
+  }
+
+  /**
+     * creates a string from snmpData to pass to influxDB
+     * @param {snmpData} object 
+     * @returns string object bodyData, to pass to body for influxDB
+     */
+  formatForInflux(snmpData){
+    logger.info('Formatted SNMP data');
+    let bodyData = 'laserJet,tag=CNB9M04409 inkLevel=' + String(snmpData.inkLevel) + '\n';
+    bodyData += 'laserJet,tag=CNB9M04409 macAddy="'  + String(snmpData.macAddy) + '"' + '\n';
+    bodyData+='laserJet,tag=CNB9M04409 currentTonerLevel='+String(snmpData.currentTonerLevel)+'\n';
+    bodyData += 'laserJet,tag=CNB9M04409 memorySize=' + String(snmpData.memorySize) + '\n';
+    bodyData += 'laserJet,tag=CNB9M04409 memoryUsed=' + String(snmpData.memoryUsed) + '\n';
+    bodyData += 'laserJet,tag=CNB9M04409 pagesPrinted=' + String(snmpData.pagesPrinted) + '\n';
+    bodyData+='laserJet,tag=CNB9M04409 serialNumber="'+String(snmpData.serialNumber)+'"'+'\n';
+    bodyData += 'laserJet,tag=CNB9M04409 modelNumber="' + String(snmpData.modelNumber) + '"' + '\n';
+    bodyData += 'laserJet,tag=CNB9M04409 tonerCapacity=' + String(snmpData.tonerCapacity);
+    logger.warn('This was Sent \n' + bodyData);
```

Comment:
Make this tag=snmpData.serialNumber to scale to both printers

Reply:
fax